velox/exec/tests/TableWriteTest.cpp

```diff
@@ -264,6 +294,115 @@ TEST_F(TableWriteTest, TestASecondCommitStrategy) {
       "SELECT * FROM tmp");
 }
 
+TEST_F(TableWriteTest, distinctPartitions) {
+  int32_t numPartitions = 50;
+  int32_t numBatches = 2;
+
+  auto rowType = ROW({"p0"}, {VARCHAR()});
+  std::vector<std::string> partitionNames = {"p0"};
+
+  auto inputFilePaths = makeFilePaths(numBatches);
+  std::vector<RowVectorPtr> vectors =
+      makeBatches(numBatches, [&](int32_t batch) {
+        return makeRowVector(
+            rowType->names(),
+            {makeFlatVector<StringView>(numPartitions, [&](auto row) {
+              return StringView(fmt::format("str_{}", row));
+            })});
+      });
+  for (int i = 0; i < numBatches; i++) {
+    writeToFile(inputFilePaths[i]->path, vectors[i]);
+  }
+
+  createDuckDbTable(vectors);
+
+  auto outputDirectory = TempDirectoryPath::create();
+  auto plan = PlanBuilder()
+                  .tableScan(rowType)
+                  .tableWrite(
+                      rowType->names(),
+                      std::make_shared<core::InsertTableHandle>(
+                          kHiveConnectorId,
+                          makeHiveInsertTableHandle(
+                              rowType->names(),
+                              rowType->children(),
+                              partitionNames,
+                              makeLocationHandle(outputDirectory->path))),
+                      CommitStrategy::kNoCommit,
+                      "rows")
+                  .project({"rows"})
+                  .planNode();
+
+  auto task = assertQuery(plan, inputFilePaths, "SELECT count(*) FROM tmp");
+
+  assertQuery(
+      PlanBuilder().tableScan(rowType).planNode(),
+      makeHiveConnectorSplits(outputDirectory),
+      "SELECT * FROM tmp");
+
+  EXPECT_EQ(
```

Comment:
Can we make this verification stronger and check the directory structure expecting one sub-directory per partition (with the right name) and one file in each sub-directory (with one row in each file)?

Reply:
Previously I tried similar but couldn't achieve it with what we can get from the std filesystem library, https://en.cppreference.com/w/cpp/filesystem.

Ex., the difficulty is, for partitions a=1/b=2 and a=1/b=3, the fs::recursive_directory_iterator() gives a list of
a=1
a=1/b=2
a=1/b=2/file2
a=1/b=3
a=1/b=3/file3

I can't tell the directory a=1 apart from the rest.